### PR TITLE
New resource/data source `azurerm_data_share_dataset_kusto_database` 

### PR DESCRIPTION
**Pull request by yupwei68**
Partially fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/6480

Sub PR of : https://github.com/terraform-providers/terraform-provider-azurerm/pull/7064

=== RUN   TestAccDataSourceAzureRMDataShareDatasetKustoDatabase_basic
=== PAUSE TestAccDataSourceAzureRMDataShareDatasetKustoDatabase_basic
=== CONT  TestAccDataSourceAzureRMDataShareDatasetKustoDatabase_basic
--- PASS: TestAccDataSourceAzureRMDataShareDatasetKustoDatabase_basic (1602.41s)

=== RUN   TestAccAzureRMCloudEndpoint_basic
=== PAUSE TestAccAzureRMCloudEndpoint_basic
=== CONT  TestAccAzureRMCloudEndpoint_basic
--- PASS: TestAccAzureRMCloudEndpoint_basic (390.42s)
=== RUN   TestAccAzureRMCloudEndpoint_requiresImport
=== PAUSE TestAccAzureRMCloudEndpoint_requiresImport
=== CONT  TestAccAzureRMCloudEndpoint_requiresImport
--- PASS: TestAccAzureRMCloudEndpoint_requiresImport (400.43s)

### CODE REVIEW

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  resource_group_name = azurerm_resource_group.test.name
+
+  sku {
+    name     = "Dev(No SLA)_Standard_D11_v2"
+    capacity = 1
+  }
+}
+
+resource "azurerm_kusto_database" "test" {
+  name                = "acctestkd-%[1]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:89]: ara capitals allowed here?
```suggestion
  name                = "acctestKD-%[1]d"
```

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+resource "azurerm_data_share" "test" {
+  name       = "acctest_ds_%[1]d"
+  account_id = azurerm_data_share_account.test.id
+  kind       = "InPlace"
+}
+
+resource "azurerm_kusto_cluster" "test" {
+  name                = "acctestkc%[3]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:78]: ara capitals allowed here?
```suggestion
  name                = "acctestKC%[3]d"
```

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  name                = "acctest-dsa-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_data_share" "test" {
+  name       = "acctest_ds_%[1]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:72]: ara capitals allowed here?
```suggestion
  name       = "acctest_DS_%[1]d"
```

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-datashare-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_data_share_account" "test" {
+  name                = "acctest-dsa-%[1]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:63]: ara capitals allowed here?
```suggestion
  name                = "acctest-DSA-%[1]d"
```

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func testAccAzureRMDataShareDataSetKustoDatabase_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-datashare-%[1]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:58]: we should start this with
```suggestion
  name     = "acctestRG-datashare-%[1]d"
```

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+)
+
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:10]: COuld we add an update test?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the pr @yupwei68 - overall this looks good aside from some comments i've left inline

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go (hunk 2 above)
Comment (yupwei68) [reply azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:78]: Not yet. Uppercase is not allowed in `azurerm_kusto_cluter` `name`.

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go (hunk 6 above)
Comment (yupwei68) [reply azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go:10]: Sorry, all fields are forcenew. There's no update method.

**Comment by yupwei68** [comment]
@katbyte Thanks for your comments. Changes have been pushed. Please continue reviewing.

**Review by katbyte — APPROVED** [review]
Thanks, LGTM now 👍

**Pull request merged**